zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -401,12 +428,15 @@ where
                     return Ok(Proposal {
                         transaction_request,
                         transparent_inputs: vec![],
-                        sapling_inputs,
+                        sapling_inputs: NonEmpty::from_vec(sapling_inputs).map(|notes| {
```

Comment:
I consciously didn't make this change, because there's not really any point; `Option<NonEmpty>` is isomorphic to `Vec` and transparent inputs don't require an anchor height, so from a type-oriented-design perspective `Vec` is the correct type; we could add a newtype around it but I don't think there's any benefit there.

Reply:
The reason I raised this question, is that I wonder whether there are things like `anchor_height` that we might want to specifically associate with the presence of transparent inputs. I don't know of any off the top of my head, but it seems worth considering.

Reply:
I mean, if we come up with a reason that we need to do this in the future, that's fine, but I don't think we need an issue for "consider this" - I *have* considered it. :)